src/Facebook/FacebookClient.php

```diff
@@ -162,7 +162,11 @@ public function getBaseGraphUrl($postToVideoUrl = false)
      */
     public function prepareRequestMessage(FacebookRequest $request)
     {
-        $postToVideoUrl = $request->containsVideoUploads();
+        $pathElements = explode('/', $request->getUrl());
+        $lastInPath = end($pathElements);
+        $postToVideoUrl = $request->containsVideoUploads() ||
+            ($lastInPath == 'videos' && $request->getMethod() === 'POST');
```

Comment:
Gotcha. Would love to hear what you think of [my feedback on the API](https://github.com/facebook/facebook-php-sdk-v4/issues/453#issuecomment-110890792). With the API I proposed we wouldn't need to do any parsing-to-guess logic. :)